### PR TITLE
extended blacklisting and whitelisting with wildcard and group rules support

### PR DESCRIPTION
**Pull request by towel**
This extends configurable settings for blacklisting and whitelisting of allowed files.
- extended blacklisting (e.g. "*.exe")
- whitelisting (e.g. "owncloud_client.exe")
- group-based black-and whitelisting rules

### CODE REVIEW

**Comment by LukasReschke** [comment]
Mhm… I'm honestly not sure if that is something that we want to support.

Furthermore this is quite an additional blob of code and requires unit tests…

**Comment by LukasReschke** [comment]
Additionally I suspect that this will have a huge performance impact.

**Comment by towel** [comment]
It's one of the features our customers frequently ask for.
Let me know if the community decides to adopt it and I'll write tests.

**Comment by Xenopathic** [comment]
We use file extension filtering on our SMB shares (secondary school in the UK). It is there to stop students storing files that may compromise system security (.exe, .zip etc) or time-wasting (.mp3, video files etc). It doesn't work. If rules like this are in place, people *will* find a way around them, even if the rules are very permissive and are there to protect the users, not hinder them (imagine a virus posing as a banned file, but the user *really* wants to use the file - they will change the file extension fully knowing something bad might happen).

TL;DR: IMHO, file extension filtering is completely ineffective.

File: config/config.sample.php [hunk 1]
```diff
@@ -856,3 +856,35 @@
...
+	),
+),
+
+'groups_whitelisted_files' => array(
+    'admin' => array(
+    ),
+),
+
+/* these blacklisted groups won't be proposed for sharing */
+'share_blacklisted_groups' => array(
```
Comment (LukasReschke) [inline config/config.sample.php]: That's completely unrelated to this change. Please create another PR and always create a PR against master and not some stable branch.

**Comment by LukasReschke** [comment]
Agreed as well as no reply for 28 days. Thus this PR is stale and I'm closing it for now.

It can get any time restored as the original branch still exists.

**Comment by chrisjscott** [comment]
Would love to see this happen.
